contracts/src/tests/test_sequencer_uptime_feed.cairo

```diff
@@ -0,0 +1,107 @@
+use starknet::ContractAddress;
+use starknet::EthAddress;
+use starknet::contract_address_const;
+use starknet::class_hash::class_hash_const;
+use starknet::class_hash::Felt252TryIntoClassHash;
+use starknet::syscalls::deploy_syscall;
+use starknet::testing::set_caller_address;
+use starknet::testing::set_contract_address;
+
+use array::ArrayTrait;
+use traits::Into;
+use traits::TryInto;
+use option::OptionTrait;
+use core::result::ResultTrait;
+
+use chainlink::emergency::sequencer_uptime_feed::SequencerUptimeFeed;
+use chainlink::ocr2::aggregator_proxy::AggregatorProxy;
+use chainlink::tests::test_ownable::should_implement_ownable;
+use chainlink::tests::test_access_controller::should_implement_access_control;
+
+// NOTE: move to separate interface file once we can directly use the trait
+#[abi]
+trait ISequencerUptimeFeed {
+    fn l1_sender() -> EthAddress;
+    fn set_l1_sender(address: EthAddress);
+}
+
+fn setup() -> (ContractAddress, ContractAddress, ISequencerUptimeFeedDispatcher) {
+    let account: ContractAddress = contract_address_const::<777>();
+    set_caller_address(account);
+
+    // Deploy seqeuencer uptime feed
+    let mut calldata = ArrayTrait::new();
+    calldata.append(0); // initial status
+    calldata.append(account.into()); // owner
+    let (sequencerFeedAddr, _) = deploy_syscall(
```

Comment:
I don't think there's a necessary requirement to deploy the contract. I think it's only necessary when we're expecting to do contract to contract interactions (calling one contract that calls another contract) or need to manage multiple contracts of the same type. In my opinion it adds some unnecessary complexity when not necessary (as opposed to just initializing the contract as a singleton). Just a nit though.

Reply:
yep i try avoid deployment where possible, but in this case the access control and ownable tests require a deployed contract to interact with a contract through IOwnableDispatcher for example